client/lib/upgrades/actions/checkout.js

```diff
@@ -29,6 +35,21 @@ function setNewCreditCardDetails( options ) {
 	} );
 }
 
+function submitFreeTransaction( site, cartItem, onComplete ) {
```

Comment:
Also I'm not sure `lib/upgrades/actions/` is the best place to put our new code. It seems it defines Flux actions only.